examples/federation-workshop/cluster-1/inspector/Identity.yaml

```diff
@@ -13,6 +13,7 @@ kind: ClusterSPIFFEID
 metadata:
   name: vsecm-inspector
 spec:
+  federatesWith: ["cluster2.demo"]
```

Comment:
You need to explicitly specify the federation relationship in `ClusterSPIFFEID`.

That’s also good security.

---

But that also makes me think of future issues about creating a `VSecMId` abstraction over ClusterSPIFFEID.


When our use cases get more complex we’ll end up utilizing SPIRE primitives such as `ClusterSPIFFEID` (which we use already) and `ClusterFederatedTrustDomain` (which we’ll have to use for federated deployments.

My initial thought was to create something like


```
apiVersion: vsecm.com/v1alpha1
kind: VSecMId
metadata:
  name: vsecm-sentinel
spec:
  matchLabels:
    app.kubernetes.io/name: vsecm-sentinel
    app.kubernetes.io/part-of: vsecm-system
  namespace: "vsecm-system"
  serviceAccount: "vsecm-sentinel"
```

which is much cleaner than

```
apiVersion: spire.spiffe.io/v1alpha1
kind: ClusterSPIFFEID
metadata:
  name: vsecm-sentinel
spec:
  spiffeIDTemplate: "spiffe://cluster1.demo\
    /workload/vsecm-sentinel\
    /ns/{{ .PodMeta.Namespace }}\
    /sa/{{ .PodSpec.ServiceAccountName }}\
    /n/{{ .PodMeta.Name }}"
  podSelector:
    matchLabels:
      app.kubernetes.io/name: vsecm-sentinel
      app.kubernetes.io/part-of: vsecm-system
  workloadSelectorTemplates:
  - "k8s:ns:vsecm-system"
  - "k8s:sa:vsecm-sentinel"
```

But,  for example, when we need to add "federation" into the mix, the ID
will need to change like this:

```
apiVersion: vsecm.com/v1alpha1
kind: VSecMId
federatesWith: ["cluster2.demo"] // <- new
metadata:
  name: vsecm-sentinel
spec:
  matchLabels:
    app.kubernetes.io/name: vsecm-sentinel
    app.kubernetes.io/part-of: vsecm-system
  namespace: "vsecm-system"
  serviceAccount: "vsecm-sentinel"
```

And then, someone will want to give arbitrary spiffeids to workloads regardless of their names (one reason could be to intercept the secrets of another workload for diagnostic/debugging purposes)

And sooner than later we might end up reinventing `ClusterSPIFFEID` and also creating additional k8s operators and controllers to reconcile our abstraction over ClusterSPIFFEID abstractions.

That feels like a lot of extra work with little-to-no additional benefit.
Plus it’ll be increased footprint because we’ll have to deploy our custom operators to manage all these too. And also additional security boundary to protect (because then we own the security of our operator(s) as well)


So I’m inclined to close the `VSecmId`-related tasks as WONTFIX.

We can discuss this in our next sync-up too.

cc: @abhishek44sharma @ats0stv @farhan-pasha